### ISSUE TITLE
`**kwargs` with `Unpack` loses type information for values inside the function.

### ISSUE THREAD

**Issue by rayansostenes** [microsoft/pyright#8089]
I think I found a bug when using `Unpack` with `**kwargs`, the type is correctly inferred when calling the function, but inside the function `**kwargs` becomes `dict[str, object]`.

```python
from typing import TypedDict,  Unpack, reveal_type, assert_type

class TestDict(TypedDict):
    name: list[str]
    age: list[str]

def f(**kwargs: Unpack[TestDict]):
    for k, v in kwargs.items():
        reveal_type(k) # Type of "k" is "str"
        reveal_type(v) # Type of "v" is "object"
        assert_type(v, list[str]) # "assert_type" mismatch: expected "list[str]" but received "object"
    reveal_type(kwargs.keys()) # Type of "kwargs.keys()" is "dict_keys[str, object]"
    reveal_type(kwargs.values()) # Type of "kwargs.values()" is "dict_values[str, object]"

f(name=[1, 2], age=["a"])
# Correctly errors with:
# 
# Argument of type "list[int]" cannot be assigned to parameter "name" of type "list[str]" in function "f"
#   "Literal[1]" is incompatible with "str"
#   "Literal[2]" is incompatible with "str"

```

[Playground Link](https://pyright-play.net/?pyrightVersion=1.1.317&pythonVersion=3.12&strict=true&locale=en&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgCYAiSAxjADRRQCqKCAhlQNb0jEBuxrADYB9eAmL1WAZynE8okgChFVQdKmFiUmJRoAKIuIrUYASgBcihgxSsIxc1EFJtAbW0gAuleus0DpxcYdxgvZVJiYChgPQAqWI4Ad1YQNClHZjZOVwItHRNPCx8GYFwoLiheTBRy5NSpADokGGIIKT0i6y6oHn4hBXE9DlMoAGJCEigwKIAiDhnMDRmPGeLu3oERMWI9XhHxw2Ip2d4FlygZsAAjACtiGlXu3xk5GAGd3npnNw9CsYv1K93gsIC4IKwYFQABaOYgAD3ENDIF2%2BwV%2BCyuAFd8DwqMQkPxSBdrncHmsNv1tkM6mkGhxiHB2qZ9hNxMcLkkUrT6YyOmclqQTMIeVIQiB6CT7jBPI9rBStiRqVzGrwhJitB0WYd2XMaSq1RrTPyLoKaMJVYJ1aKPBLblKZcoYrZ7ABeVwARnoACZPJJ-G6ZqwZoVFOMAMK4XEwQRwKBycAgDSJZow0NQNMAQVSmPsKHw01gkxmqNcqGlCyorBQKDA%2BCuR3USDQKGRMDAUDYIDsxBaIAuzuICwL2xRQTFMuq0UxKBoSDANRmwFW4wYMwAMs05EIPRPzqgqJA2DAkFdBEdkzAoRcVmnVxve9ufcb94eISez1AL1flqFVoogA)

**Comment by erictraut** [comment]
Pyright's behavior here is correct. A `TypedDict` is a structural type, and it doesn't guarantee that there are no other keys present. There's a draft [PEP 728](https://peps.python.org/pep-0728/) that proposes to add support for "closed" TypedDict definitions where no additional items can be present, but that spec hasn't yet been finalized. Pyright does have experimental support for this PEP if you want to give it a try. You'd need to set `enableExperimentalFeatures` to true in your config file. Then you'd need to change your definition of `TestDict` as follows:

```python
class TestDict(TypedDict, closed=True):
    name: list[str]
    age: list[str]
```

For traditional (non-closed) TypedDicts, a type checker must assume that additional keys are present and that the values associated with those keys can be any type. That's why `reveal_type(v)` reveals `object`.
